src/action-ros-ci.ts

```diff
@@ -362,6 +364,9 @@ async function run_throw(): Promise<void> {
 		options.env = {
 			...process.env,
 			COLCON_DEFAULTS_FILE: colconDefaultsFile,
+			ROS_VERSION: targetRos1Distro ? "1" : "2",
+			ROS_PYTHON_VERSION:
+				targetRos1Distro && targetRos1Distro != "noetic" ? "2" : "3",
```

Comment:
You should add these new variables to the environment without any condition (without `colconDefaultsFile !== ""`), so you need to move them above, outside of the `if`.

Reply:
Ah, exactly. I overlooked. Thank you!

Reply:
I'm not at my computer so this is difficult to write, but it should be like this:

```ts
	const options: im.ExecOptions = {
		cwd: rosWorkspaceDir,
                env: {
                    ...process.env,
                    ROS_VERSION: targetRos1Distro ? "1" : "2",
	            ROS_PYTHON_VERSION:
		        targetRos1Distro && targetRos1Distro != "noetic" ? "2" : "3",
                },
	};
	if (colconDefaultsFile !== "") {
		options.env = {
			...options.env,
			COLCON_DEFAULTS_FILE: colconDefaultsFile,
                }
        }
```

Reply:
@christophebedard All tests have passed! (Of course, it's mainly thanks to you! :smile: )
Could you review, please? :pray:

Reply:
Also, I found another problem. Sent a PR for that. #701